nativelib/src/main/resources/cre2.cpp

```diff
@@ -0,0 +1,297 @@
+/*
+  https://github.com/marcomaggi/cre2 (0.3.0)
+
+  Source  file  for  CRE2, a  C   language  wrapper  for RE2:  a   regular
+  expressions library by Google.
+
+  Copyright (c) 2012 Marco Maggi <marco.maggi-ipsu@poste.it>
+  Copyright (c) 2011 Keegan McAllister
+  All rights reserved.
+
+  Redistribution and  use in source and binary  forms, with or
+  without  modification,  are   permitted  provided  that  the
+  following conditions are met:
+
+  1.  Redistributions of  source  code must  retain the  above
+     copyright  notice,  this   list  of  conditions  and  the
+     following disclaimer.
+
+  2. Redistributions  in binary form must  reproduce the above
+     copyright  notice,  this   list  of  conditions  and  the
+     following  disclaimer in  the documentation  and/or other
+     materials provided with the distribution.
+
+  3.  Neither the  name of  the author  nor the  names  of his
+     contributors may  be used to endorse  or promote products
+     derived from this software without specific prior written
+     permission.
+
+  THIS  SOFTWARE  IS PROVIDED  BY  THE  COPYRIGHT HOLDERS  AND
+  CONTRIBUTORS   ``AS  IS''   AND  ANY   EXPRESS   OR  IMPLIED
+  WARRANTIES,  INCLUDING,  BUT  NOT  LIMITED TO,  THE  IMPLIED
+  WARRANTIES OF  MERCHANTABILITY AND FITNESS  FOR A PARTICULAR
+  PURPOSE ARE  DISCLAIMED.  IN NO  EVENT SHALL THE  AUTHORS OR
+  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+  SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT
+  NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+  LOSS  OF USE,  DATA, OR  PROFITS; OR  BUSINESS INTERRUPTION)
+  HOWEVER CAUSED  AND ON ANY  THEORY OF LIABILITY,  WHETHER IN
+  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+  OTHERWISE)  ARISING  IN ANY  WAY  OUT  OF  THE USE  OF  THIS
+  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
+
+// Scala-Native version removes unused functions
+
+#include <re2/re2.h>
+#include <re2/set.h>
+#include "cre2.h"
+
+#include <cstdlib>
+#include <cstdio>
+#include <vector>
+#include <map>
+
+/** --------------------------------------------------------------------
+ ** Options objects.
+ ** ----------------------------------------------------------------- */
+
+/* Cast    the  pointer    argument   "opt"   to   a   pointer   of   type
+   "RE2::Options*". */
+#define TO_OPT(opt) (reinterpret_cast<RE2::Options *>(opt))
+
+/* Allocate and return a new options object. */
+cre2_options_t * cre2_opt_new(void) {
```

Comment:
The problem is that ours `cre2` is a modified version of the original that we inline into your binary without you knowing about it. I think it's going to be way less confusing to keep it prefixed. Moreover, this would also let us change the implementation later on without being locked into having to support re2 forever.